web/src/js/components/Authentication/EnterEmailForm.js

```diff
@@ -0,0 +1,217 @@
+        {props.buttonText}
+      </Button>
+    </span>
+  </Paper>
+)
+
+class EnterEmailForm extends React.Component {
+  constructor(props) {
+    super(props)
+    this.emailFieldRef = null
+    this.state = {
+      emailDuplicate: false,
+      otherError: false,
+      savingEmailInProgress: false,
+      verifyEmailSent: false,
+      inputEmail: null,
+    }
+  }
+
+  verifyEmailSent() {
+    this.setState({ verifyEmailSent: true })
+  }
+
+  updateUserFailure(error) {
+    console.log(error)
+    if (error.code === 'auth/requires-recent-login') {
+      replaceUrl(loginURL, {
+        next: '/newtab/account',
+        reauth: 'true',
+      })
+    } else if (error.code === 'auth/email-already-in-use') {
+      this.setState({ emailDuplicate: true })
+    }
+  }
+
+  submit() {
+    const email = this.emailFieldRef.getValue()
+    this.setState({
+      emailDuplicate: false,
+      otherError: false,
+      inputEmail: email,
+    })
+    updateUserEmail(email)
+      .then(() => this.verifyEmailSent())
+      .catch(this.updateUserFailure.bind(this))
+  }
+
+  resubmit() {
+    updateUserEmail(this.state.inputEmail)
+      .then(() => this.verifyEmailSent())
+      .catch(this.updateUserFailure)
+  }
+
+  onMutationCompleted(response) {
+    const { onCompleted } = this.props
+    this.setState({
+      savingEmailInProgress: false,
+    })
+    const data = response.setEmail
+
+    // Handle server-side validation errors.
+    if (!data.user) {
+      data.errors.forEach(err => {
+        // Email already exists
+        if (err.code === 'auth/email-already-in-use') {
+          this.setState({
+            emailDuplicate: true,
+          })
+          // Some other error
+        } else {
+          this.setState({
+            otherError: true,
+          })
+        }
+      })
+      return
+    }
+
+    // Go to the app.
+    onCompleted()
+  }
+
+  onMutationError(response) {
+    logger.error(response)
+    this.setState({
+      savingEmailInProgress: false,
+      otherError: true,
+    })
+  }
+
+  handleKeyPress(e) {
+    if (this.state.savingEmailInProgress) {
+      return
+    }
+
+    if (e.key === 'Enter') {
+      this.submit()
+    }
+  }
+
+  render() {
+    const { app } = this.props
+    return this.state.verifyEmailSent ? (
+      <PaperItem
+        title="Verify Your New Email Address"
+        text="Please check your email to verify your new address."
+        buttonText="Resend Verification"
+        buttonHandler={this.resubmit.bind(this)}
+      />
+    ) : (
+      <Paper
+        elevation={1}
+        style={{
+          padding: 24,
+          backgroundColor: '#FFF',
+        }}
+      >
+        <EmailField
+          data-test-id={'enter-email-form-email-field'}
+          ref={elem => {
+            this.emailFieldRef = elem
+          }}
+          emailDuplicate={this.state.emailDuplicate}
+          otherError={this.state.otherError}
+          onKeyPress={this.handleKeyPress.bind(this)}
```

Comment:
@jedtan Tried testing this locally and on dev (verified that this code is on dev) and when I try to submit my new email it logs me out without updating.  I think my config might be slightly off on local, can you give me steps to test?

Reply:
Hmm, I don't think I had to do anything, I still have my default test user is the "abcdefghijklmno" one - did it work for you when it was on dev?